assets/js/base/components/payment-methods/saved-payment-method-options.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useState, useRef, useCallback } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { usePaymentMethodDataContext } from '@woocommerce/base-context';
+import RadioControl from '@woocommerce/base-components/radio-control';
+
+/** @typedef {import('@woocommerce/type-defs/contexts').CustomerPaymentMethod} CustomerPaymentMethod */
+
+/**
+ * Returns the option object for a cc or echeck saved payment method token.
+ *
+ * @param {CustomerPaymentMethod} savedPaymentMethod
+ *
+ * @return {Object} An option objects to use for RadioControl.
+ */
+const getCcOrEcheckPaymentMethodOption = ( { method, expires, tokenId } ) => {
+	return {
+		value: tokenId,
+		label: sprintf(
+			__(
+				'%1$s ending in %2$s (expires %3$s)',
+				'woo-gutenberg-product-blocks'
+			),
+			method.brand,
+			method.last4,
+			expires
+		),
+		name: `wc-saved-payment-method-token-${ tokenId }`,
+	};
+};
+
+/**
+ * Returns the option object for any non specific saved payment method.
+ *
+ * @param {CustomerPaymentMethod} savedPaymentMethod
+ *
+ * @return {Object} An option objects to use for RadioControl.
+ */
+const getDefaultPaymentMethodOptions = ( { method, tokenId } ) => {
+	return {
+		value: tokenId,
+		label: sprintf(
+			__( 'Saved token for %s', 'woo-gutenberg-products-block' ),
+			method.gateway
+		),
+		name: `wc-saved-payment-method-token-${ tokenId }`,
+	};
+};
+
+const SavedPaymentMethodOptions = ( { onSelect } ) => {
```

Comment:
Is this component specific to Stripe or should it work with multiple payment method providers (i.e. extensions)? 

My understanding is that it's a saved stripe token for a shopper's specific card.

If this is specific to stripe, should this be in a stripe subfolder?

Reply:
> If this is specific to stripe, should this be in a stripe subfolder?

No. It's modelled after functionality that _exists in WooCommerce core_ currently (php side).

Reply:
Thanks for clarifying. Perhaps this is a saved "instance" of a payment method with details, e.g. "stripe + card 1234"?